src/power_grid_model/data_types.py

```diff
@@ -7,84 +7,210 @@
 have been defined and explained in this file.
 """
 
-from typing import TypeVar, Union
+from typing import TypeAlias, TypeVar, Union
 
 import numpy as np
 
-from power_grid_model.core.dataset_definitions import ComponentTypeLike  # pylint: disable=unused-import
 from power_grid_model.core.dataset_definitions import ComponentTypeVar
 
+AttributeType: TypeAlias = str
+"""
+An attribute type is a string reprenting the attribute type of a specific component.
+
+- Examples:
+
+    - "id"
+    - "u_rated"
+"""
+
+SparseDataComponentType: TypeAlias = str
+"""
+A string representing the component type of sparse data structures.
+
+Must be either "data" or "indptr".
+"""
+
```

Comment:
This is really setting things up to introduce proper typing and `Enum`s for `AttributeType`s Isn't it? Should we maybe bring up discussions about this?

Reply:
we can indeed

Reply:
we can at any time